dhall-json/json-to-dhall/Main.hs

```diff
@@ -46,9 +49,14 @@ parserInfo = Options.info
           )
 
 -- | All the command arguments and options
+data Schemas
+    =  SchemaName (Maybe Text)
+    |  SchemasFile FilePath
+    deriving Show
+
 data Options
     = Default
-        { schema     :: Maybe Text
+        { schemas    :: Schemas
```

Comment:
I think we should use a separate command-line option for specifying the schemas used to simplify the file

The reason why is that the existing singular schema argument is used to specify the expected Dhall type of the complete file; if the file does not match the declared schema then the conversion rejects the file.

These other new schemas are used for a different purpose: simplifying the generated Dhall code.  If none of them match then it's not an error.

The other reason we want to separate the two concepts is because the user is allowed to omit the old singular schema and let the compiler infer the correct Dhall type.  However, we'd still like to be able to simplify the inferred type with these latter new schemas if they happen to match against the type.

Reply:
I assumed those would be conflicting options, either the conversion match a single type, or it find the type from a list. I now see how both concepts can be used together, it would actually make the Main.hs modification simpler.